crates/scheduler/src/call/button.rs

```diff
@@ -45,7 +45,7 @@ fn register<B: Board>(mut call: SchedulerCall<B, api::register::Sig>) {
     let api::register::Params { button, handler_func, handler_data } = call.read();
     let inst = call.inst();
     let result = try {
-        let button = Id::new(*button as usize).ok_or(Trap)?;
+        let button = Id::new(*button as usize)?;
```

Comment:
changed this to a `map_err(|_| Trap)` as per your suggestion for splitting this PR.

BTW: I've been using `scripts/ci.sh` to test the build as `rust-analyzer` wasn't giving me good type checks for some of these files. Is this a known issue or do you think I have something wrong with my setup?

Reply:
Each crate should have a `./test.sh` that you can use to test that crate. This is much faster than running `./scripts/ci.sh`.

Regarding `rust-analyzer`, you need a way to configure the target and features. By default it probably uses x86-64 and the default features. But that's not always what you want. If you use Emacs I can provide you my setup. Otherwise you have to check in your editor how to configure rust-analyzer.